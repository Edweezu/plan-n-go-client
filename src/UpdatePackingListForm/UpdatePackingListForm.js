import React from 'react'
import TripsContext from '../TripsContext';
import config from '../config'
import TokenService from '../services/token-service'
import { confirmAlert } from 'react-confirm-alert'

class UpdatePackingListForm extends React.Component {

    static contextType = TripsContext

   

   
    handleDeleteItem = () => {
        let { tripid, itemid } = this.props

        fetch(`${config.API_ENDPOINT}/trips/${tripid}/packing_list/${itemid}`, {
            method: 'DELETE',
            headers: {
                'content': 'application-json',
                'authorization': `bearer ${TokenService.getAuthToken()}`
            }
        })
        .then(res => {
            if (!res.ok) {
                return res.json().then(e => Promise.reject(e))
            }
        })
        .then(data => {
            this.context.deleteItem(itemid)
        })
        .catch(err => {
            console.error(err)
        })
    }

    handleDeleteForm = () => {
        confirmAlert({
            title: '',
            message: 'Are you sure you want to delete this?',
            buttons: [
                {
                    label: 'Yes',
                    onClick : () => {
                        this.handleDeleteItem()
                    }
                },
                {
                    label: 'No',
                    onClick : () => {
                        this.setState({
                            error: null
                        })
                    }
                }
            ]
        })
    }
    



    render () {

        return (
            <span>
                <i onClick={this.handleDeleteForm}className="far fa-trash-alt"></i>
                {/* <button onClick={this.handleDeleteItem}>
                    Delete
                </button> */}
            </span>  
        )
    }
}

export default UpdatePackingListForm