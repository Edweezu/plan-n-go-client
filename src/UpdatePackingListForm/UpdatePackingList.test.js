import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import UpdatePackingListForm from './UpdatePackingListForm'


describe('UpdatePackingListForm component', () => {
    const props = {
        tripid: 1,
        itemid: 1
    }

    it('renders the given component', () => {
        const wrapper = shallow(<UpdatePackingListForm />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it(`renders the component's given props`, () => {
        const wrapper = shallow(<UpdatePackingListForm {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})