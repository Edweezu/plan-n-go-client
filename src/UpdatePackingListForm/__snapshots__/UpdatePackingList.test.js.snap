// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdatePackingListForm component renders the component's given props 1`] = `
<div
  className="btn-div"
>
  <i
    className="fas fa-check"
    onClick={[Function]}
  />
  <i
    className="far fa-trash-alt"
    onClick={[Function]}
  />
</div>
`;

exports[`UpdatePackingListForm component renders the given component 1`] = `
<div
  className="btn-div"
>
  <i
    className="fas fa-check"
    onClick={[Function]}
  />
  <i
    className="far fa-trash-alt"
    onClick={[Function]}
  />
</div>
`;
